src/lfortran/semantics/ast_body_visitor.cpp

```diff
@@ -273,6 +273,18 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                 if (!ASRUtils::is_character(*a_form_type)) {
                         throw SemanticError("`form` must be of type, Character or CharacterPointer", x.base.base.loc);
                 }
+            } else {
+                const std::unordered_set<std::string> unsupported_args {"iostat", "iomsg", "err", "blank", "access", \
+                                                                            "recl", "fileopt", "action", "position"};
+                if (unsupported_args.find(m_arg_str) == unsupported_args.end()) {
+                    throw SemanticError("Invalid argument `" + m_arg_str + "` supplied", x.base.base.loc);
+                } else {
+                    diag.semantic_warning_label(
+                        "Argument `" + m_arg_str + "` isn't supported yet",
+                        {x.base.base.loc},
```

Comment:
Let's use something like:
```suggestion
                        {kwarg.base.base.loc},
```
So that it highlights exactly the argument that we need, instead of the whole thing.

If this works, then the same for all the other errors.

Reply:
when I try `kwarg.loc`, I get:
```bash
(lf1)  ✘ gxyd@Gauravs-MacBook-Air  ~/OpenSource/lfortran   invalid_kwarg-3207 ±  ./src/bin/lfortran tests/warnings/open_notimplemented_kwarg1.f90
warning: Argument `action` isn't supported yet
 --> tests/warnings/open_notimplemented_kwarg1.f90:1:1
  |
1 | PROGRAM open_notimplemented_kwargs
  | ^ ignored for now

```

I noted the same issue somewhere else as well, which I believe is incorrect. I believe kwarg's aren't being assigned `loc` correctly (they are just set to `0, 0` for first, last)

Reply:
I'm looking at `asdl_cpp.py` for where to assign `loc` for kwargs correctly.

Reply:
Ok, keep the original code for now, let's merge this PR and fix it up later.

Can you open up an issue for the kward loc problem?

Reply:
Done, I've opened an isssue here: https://github.com/lfortran/lfortran/issues/3295